Attachment is not a function LTI error fixed

wsmirnow opened the pull request:
In some cases the LTI series tool fails to render the page for a specific series. Instead of the list of episodes you will see an error Message `Error:  e.mediapackage.attachments.attachment.map is not a function`. I've tracked it down to one episode where the `mediapackage.attachments.attachment` is not an array as expected but an attachment object. This PR fixes the bug.


### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] be against the correct branch (features can only go into develop)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

Arnei commented:
To test this, I:
- Uncommented every operation in the fast workflow that has anything to do with "preview".
- Uploaded an event using the fast workflow WITHOUT a series.

This should leave you with an event with exactly one attachment.

wsmirnow committed:
Fixed error 'e.mediapackage.attachments.attachment.map is not a function' in the LTI series view

Under some conditions an engage pulication can contain only one attachment. The search service doesn't wrap one attachment element as an array but leave it as an json object. At this point the LTI series view fails with error above.

wsmirnow commented:
Conflict resolved, rebased and improved commit message.

KatrinIhler commented:
Since Arne tested this, but cannot merge it, I'm gonna go ahead and do that. :)
